src/main/flight/autopilot.c

```diff
@@ -22,38 +22,142 @@
+    pt1FilterInit(&efAxis->velocityLpf, ap.pt1Gain); // Clear and initialise the filters
+    pt1FilterInit(&efAxis->accelerationLpf, ap.pt1Gain);
+    efAxis->isStopping = true; // Enter starting 'phase'
+    efAxis->integral = 0.0f;
+}
+
+void resetPt3UpsampleFilters(void)
+{
+    pt3FilterInit(&ap.upsample[AI_ROLL], ap.upsampleCutoff);
+    pt3FilterInit(&ap.upsample[AI_PITCH], ap.upsampleCutoff);
+}
+
+void resetPositionControl(gpsLocation_t *initialTargetLocation)
+{
+    // from pos_hold.c when initiating position hold at target location
+    currentTargetLocation = *initialTargetLocation;
+    ap.sticksActive = false;
+    // set sanity check distance according to groundspeed at start
+    ap.sanityCheckDistance = gpsSol.groundSpeed > 500 ? gpsSol.groundSpeed * 2.0f : 1000.0f;
+    resetPositionControlEFParams(&ap.efAxis[EW]);
+    resetPositionControlEFParams(&ap.efAxis[NS]);
+    resetPt3UpsampleFilters(); // clear anything from previous iteration
+}
+
+void initializeEfAxisFilters(earthFrame_t *efAxis, float filterGain) {
+    pt1FilterInit(&efAxis->velocityLpf, filterGain);
+    pt1FilterInit(&efAxis->accelerationLpf, filterGain);
+}
+
+void autopilotInit(void)
 {
-    altitudePidCoeffs.Kp = config->altitude_P * ALTITUDE_P_SCALE;
-    altitudePidCoeffs.Ki = config->altitude_I * ALTITUDE_I_SCALE;
-    altitudePidCoeffs.Kd = config->altitude_D * ALTITUDE_D_SCALE;
-    altitudePidCoeffs.Kf = config->altitude_F * ALTITUDE_F_SCALE;
+    ap.sticksActive = false;
+    ap.maxAngle = autopilotConfig()->max_angle;
+    altitudePidCoeffs.Kp = autopilotConfig()->altitude_P * ALTITUDE_P_SCALE;
+    altitudePidCoeffs.Ki = autopilotConfig()->altitude_I * ALTITUDE_I_SCALE;
+    altitudePidCoeffs.Kd = autopilotConfig()->altitude_D * ALTITUDE_D_SCALE;
+    altitudePidCoeffs.Kf = autopilotConfig()->altitude_F * ALTITUDE_F_SCALE;
+    positionPidCoeffs.Kp = autopilotConfig()->position_P * POSITION_P_SCALE;
+    positionPidCoeffs.Ki = autopilotConfig()->position_I * POSITION_I_SCALE;
+    positionPidCoeffs.Kd = autopilotConfig()->position_D * POSITION_D_SCALE;
+    positionPidCoeffs.Kf = autopilotConfig()->position_A * POSITION_A_SCALE; // Kf used for acceleration
+    // initialise filters with approximate filter gain
+    ap.upsampleCutoff = pt3FilterGain(UPSAMPLING_CUTOFF_HZ, 0.01f); // 5Hz, assuming 100Hz task rate
+    resetPt3UpsampleFilters();
+    // Initialise PT1 filters for earth frame axes NS and EW
+    ap.pt1Cutoff = autopilotConfig()->position_cutoff * 0.01f;
+    ap.pt1Gain = pt1FilterGain(ap.pt1Cutoff, 0.1f); // assume 10Hz GPS connection at start
+    initializeEfAxisFilters(&ap.efAxis[EW], ap.pt1Gain);
+    initializeEfAxisFilters(&ap.efAxis[NS], ap.pt1Gain);
 }
 
 void resetAltitudeControl (void) {
     altitudeI = 0.0f;
 }
 
-void altitudeControl(float targetAltitudeCm, float taskIntervalS, float verticalVelocity, float targetAltitudeStep) {
-
+void altitudeControl(float targetAltitudeCm, float taskIntervalS, float targetAltitudeStep)
```

Comment:
Well... I try to please @KarateBrot who suggested that all 'autopilot' type functions should be put in an 'autopilot' file, which is always available, so that they can be called by 'client' functions.
Hence both gps_rescue and alt_hold access and use this function as part of their 'autopilot' behaviour.
If we had other functions (eg perhaps fly to point, or geofence, or altitude limiting), they can call this function from within autopilot.c.
I feel that 'no place is perfect' but think that this is 'as good a place as any'.
Hence I perhaps think - without strong feelings either way - that this is the right place.

Also `USE_ALT_HOLD_MODE` is a build requirement, but `autopilot` is a 'core' code block.  And I don't want someone having to include `USE_ALT_HOLD_MODE` for GPS Rescue to have access to the altitude control code.